### PR TITLE
Model

### PR DESCRIPTION
**Pull request by vladimirPerehudov**
create entity

### CODE REVIEW

File: src/main/java/com/example/forumbackend/model/Post.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
+
+    private String post_text;
+
+    private Time post_time;
+
+    @OneToOne
```
Comment (dmitryPeregudov) [inline src/main/java/com/example/forumbackend/model/Post.java:20]: wrong relationship

File: src/main/java/com/example/forumbackend/model/Post.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    private User user;
```
Comment (dmitryPeregudov) [inline src/main/java/com/example/forumbackend/model/Post.java:21]: How admin can hide message

File: src/main/java/com/example/forumbackend/model/Role.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+@Data
+@Entity
+public class Role {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
+
+    private String role_name;
```
Comment (dmitryPeregudov) [inline src/main/java/com/example/forumbackend/model/Role.java:15]: If we have only two roles- do we need a public constructor?? better approach is to create a private one, add two static roles and add a method that returns default role and a method that returns role based on role name

File: src/main/java/com/example/forumbackend/model/User.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@Entity
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
+
+    private String login;
+
+    private String password;
```
Comment (dmitryPeregudov) [inline src/main/java/com/example/forumbackend/model/User.java:17]: this should pe encrypted and should have write-only access

File: src/main/java/com/example/forumbackend/model/User.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    private String email;
+
+    private String first_name;
+
+    private String last_name;
+
+    @ManyToOne(optional = false, cascade = CascadeType.ALL)
```
Comment (dmitryPeregudov) [inline src/main/java/com/example/forumbackend/model/User.java:25]: You should not use CascadeType.ALL here

File: src/main/java/com/example/forumbackend/model/User.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package com.example.forumbackend.model;
+
+import lombok.Data;
+
+import javax.persistence.*;
+
+@Data
+@Entity
+public class User {
```
Comment (dmitryPeregudov) [inline src/main/java/com/example/forumbackend/model/User.java:9]: how admin can ban your user

File: pom.xml [hunk 7]
```diff
@@ -60,6 +60,13 @@
             <artifactId>jjwt</artifactId>
             <version>0.9.1</version>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/com.h2database/h2 -->
```
Comment (dmitryPeregudov) [inline pom.xml:63]: we don't need this comment

File: src/main/resources/SQL_inputs.txt [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+CREATE TABLE Post
+(
+id INT PRIMARY KEY AUTO_INCREMENT,
+post_text VARCHAR NOT NULL,
+post_time TIME,
+user_id INT,
+FOREIGN KEY (user_id) REFERENCES User (id) ON DELETE CASCADE
+);
+
+select * from ROLE;
```
Comment (dmitryPeregudov) [inline src/main/resources/SQL_inputs.txt:32]: remove this

File: src/main/resources/SQL_inputs.txt [hunk 9]
```diff
@@ -0,0 +1,34 @@
+drop table USER;
+drop table ROLE;
+drop table POST;
+
+CREATE TABLE Role
+(
+id INT PRIMARY KEY AUTO_INCREMENT,
+role_name varchar NOT NULL UNIQUE
+);
+
```
Comment (dmitryPeregudov) [inline src/main/resources/SQL_inputs.txt:10]: we should add some roles after we create a table. Please write script

**Comment by vladimirPerehudov** [comment]
corrected notes

**Pull request merged**